orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy

```diff
@@ -44,10 +45,15 @@ class UpsertApplicationTask extends AbstractFront50Task {
     } else {
       log.info("Creating application (name: ${application.name})")
       front50Service.create(application)
+      if (application?.permission?.permissions == null) {
```

Comment:
You can probably groovify this with `if (!application?.permission?.permissions)`

Reply:
I want this condition to be true for an empty map - `null` should mean no change to the permissions, whereas `[:]` should mean an empty set of permissions.

Reply:
It'll also be true if `application` or `permission` are `null` when written like this -- is that intended?

Reply:
Yes for `permission`, no for `application`, so I'll take out the null check for application.